src/org/rascalmpl/parser/uptr/recovery/CaseInsensitiveLiteralMatcher.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2022, NWO-I Centrum Wiskunde & Informatica (CWI)
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
+
+ package org.rascalmpl.parser.uptr.recovery;
+
+public class CaseInsensitiveLiteralMatcher implements InputMatcher {
+	private final int[] chars;
+    private final int[] altChars;
+
+    public CaseInsensitiveLiteralMatcher(String literal) {
+        int length = literal.length();
+		chars = new int[length];
+		altChars = new int[length];
+		for(int i = 0; i<length; i++){
+			int character = literal.codePointAt(i);
+            chars[i] = character;
+			switch (Character.getType(character)) {
```

Comment:
there is also `Character.isLowerCase` which I think is a bit more accurate for unicode codepoints (as there are apparantly 2 kinds of lowercase)

Reply:
The code is now using `isLowerCase`/`isUpperCase`. I also replaced the code in `CaseInsensitiveLiteralStackNode` that does the same.